kong/admin_gui/init.lua

```diff
@@ -0,0 +1,56 @@
+local meta       = require "kong.meta"
+
+local _M = {}
+
+-- return first listener matching filters
+local function select_listener(listeners, filters)
+  for _, listener in ipairs(listeners) do
+    local match = true
+    for filter, value in pairs(filters) do
+      if listener[filter] ~= value then
+        match = false
+      end
+    end
+    if match then
+      return listener
+    end
+  end
+end
+
+local function prepare_variable(variable)
+  if variable == nil then
+    return ""
+  end
+
+  return tostring(variable)
+end
+
+function _M.generate_kconfig(kong_config)
+  local api_listen = select_listener(kong_config.admin_listeners, {ssl = false})
+  local api_port = api_listen and api_listen.port
+  local api_ssl_listen = select_listener(kong_config.admin_listeners, {ssl = true})
+  local api_ssl_port = api_ssl_listen and api_ssl_listen.port
+
+  local configs = {
+    ADMIN_GUI_URL = prepare_variable(kong_config.admin_gui_url),
+    ADMIN_GUI_PATH = prepare_variable(kong_config.admin_gui_path),
+    ADMIN_API_URL = prepare_variable(kong_config.admin_gui_api_url),
+    ADMIN_API_PORT = prepare_variable(api_port),
+    ADMIN_API_SSL_PORT = prepare_variable(api_ssl_port),
+    KONG_VERSION = prepare_variable(meta.version),
+    KONG_EDITION = "community",
```

Comment:
let's use `kong.version` replace `community` to avoid CE and EE merge conflict?

Reply:
Have an offline chat with @oowl, we agree to unify this later